### PR TITLE
Add support for template omit

### PR DESCRIPTION
**Pull request by jkupferer**
- Similar to Ansible `omit`, this allows templates to generate values that should be removed from the final object.

### CODE REVIEW

File: operator/poolboy_templating.py [hunk 1]
```diff
@@ -154,17 +156,43 @@ def jinja2process(template, template_style='jinja2', variables={}):
...
+            key: __recursive_strip_omit(val, omit=omit)
+            for key, val in value.items()
+            if val != omit
+        }
+    elif isinstance(value, list):
+        return [
+            __recursive_strip_omit(item, omit=omit) for item in value if item != omit
+        ]
+    elif value != omit:
+        return value
```
Comment (fridim) [inline operator/poolboy_templating.py:198]: unit tests for this function could be easy to add?
validate edge cases too, like None, [], {}, etc

File: operator/poolboy_templating.py [hunk 2]
```diff
@@ -154,17 +156,43 @@ def jinja2process(template, template_style='jinja2', variables={}):
...
         }
     elif isinstance(template, list):
         return [
-            recursive_process_template_strings(item, template_style=template_style, variables=variables)
+            __recursive_process_template_strings(item, omit=omit, template_style=template_style, variables=variables)
             for item in template
         ]
     elif isinstance(template, str):
-        return jinja2process(template=template, template_style=template_style, variables=variables)
+        return jinja2process(omit=omit, template=template, template_style=template_style, variables=variables)
```
Comment (fridim) [inline operator/poolboy_templating.py:182]: even if they are named parameter, i would add the new one at the end

**Review by fridim — APPROVED** [review]

File: operator/poolboy_templating.py (hunk 1 above)
Comment (jkupferer) [reply operator/poolboy_templating.py:198]: Will do

File: operator/poolboy_templating.py (hunk 2 above)
Comment (jkupferer) [reply operator/poolboy_templating.py:182]: I prefer having named parameters alphabetized.

Comment (fridim) [reply operator/poolboy_templating.py:182]: Ah ok.

It's just that generally speaking, if anything uses it without the names, it can break.

```python
def bla(b=1, c=2, d=3):
    print(b, c, d)


b = 1
c = 2
d = 3
bla(b, c, d)   # prints 1 2 3
```

if you introduce `a`:

```python

def bla(a=0, b=1, c=2, d=3):
    print(b, c, d)

b = 1
c = 2
d = 3
bla(b, c, d) # prints 2 3 3
```

Comment (fridim) [reply operator/poolboy_templating.py:182]: I realized i didn't highlight the right place:

```diff
- def jinja2process(template, template_style='jinja2', variables={}):
+ def jinja2process(template, omit=None, template_style='jinja2', variables={}):
```

Comment (jkupferer) [reply operator/poolboy_templating.py]: Okay, changing the call to match the arguments and pass template as positional.

File: operator/poolboy_templating.py (hunk 1 above)
Comment (jkupferer) [reply operator/poolboy_templating.py:198]: Added unit tests.
